packages/synth.queues/src/services/kafka.queue.service.ts

```diff
@@ -0,0 +1,88 @@
+import { MultivestError, PluginManager } from '@applicature/synth.plugin-manager';
+import * as config from 'config';
+import { createReadStream, createWriteStream, KafkaConsumer } from 'node-rdkafka';
+import * as logger from 'winston';
+import { Errors } from '../errors';
+import { Message, Queue } from '../types';
+import { QueueService } from './queue.service';
+
+export class KafkaQueueService extends QueueService {
+    private producer: any;
+    private consumer: any;
+
+    constructor(pluginManager: PluginManager) {
+        super(pluginManager);
+        if (config.has('kafka.consumer')) {
+            const kafkaConsumer = config.get('kafka.global');
+            this.consumer = new KafkaConsumer(kafkaConsumer, {});
+        }
+    }
+    public async createQueue(queueName: string): Promise<Queue> {
+        if (config.has('kafka.consumer')) {
+            this.consumer = createReadStream(config.get('kafka.global'), config.get('kafka.topic'), {
+                topics: queueName
+            });
+            this.consumer.setPollInterval(100);
+        } else {
+            throw new MultivestError(Errors.KAFKA_CONFIG_NOT_FOUND);
+        }
+
+        if (config.has('kafka.producer')) {
+            this.producer = createWriteStream(config.get('kafka.global'), config.get('kafka.topic'), {
+                topics: queueName
+            });
+            this.producer.setPollInterval(100);
+        } else {
+            throw new MultivestError(Errors.KAFKA_CONFIG_NOT_FOUND);
+        }
+        return Promise.resolve({
+            uniqueTag: queueName
+        });
+    }
+    public async deleteQueue(queueUrl: string): Promise<void> {
+        this.producer = null;
+        this.consumer = null;
```

Comment:
is this action actually deletes a topic from Kafka?

Reply:
By default, for manual topic deleting, you should use `Zookeeper` (native C++ Kafka). 

https://stackoverflow.com/questions/33537950/how-to-delete-a-topic-in-apache-kafka
Here you can see, that manual deleting was supported in `1.0.0` version by adding `delete.topic.enable` to params. In actual `node-rdkafka` version, there's no possible way to set this param. So I don't know if there a possible way to make this manually.